Mathlib/Tactic/Rify.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2023 Patrick Massit. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Doll, Mario Carneiro, Robert Y. Lewis, Patrick Massot
+-/
+import Mathlib.Tactic.Qify
+import Mathlib.Data.Real.Basic
+
+/-!
+# `rify` tactic
+
+The `rify` tactic is used to shift propositions from `ℕ`, `ℤ` or `ℚ` to `ℝ`.
+
+Although less useful than its cousins `zify` and `qify`, it can be useful when your
+goal or context already involves real numbers.
+
+In the example below, assumption `hn` is about natural numbers, `hk` is about integers
+and involves casting a natural number to `ℤ`, and the conclusion is about real numbers.
+The proof uses `rify` to lift both assumptions to `ℝ` before calling `linarith`.
+```
+import Mathlib.Tactic.Linarith
+import Mathlib.Tactic.Rify
+
+example {n : ℕ} {k : ℤ} (hn : 8 ≤ n) (hk : 2 * k ≤ n + 2) :
+    (0 : ℝ) < n - k - 1 := by
+  rify at hn hk
+  linarith
+```
+-/
```

Comment:
Can you add a list of TODOs here? For instance, generalize from `ℝ` to `LinearOrderedField` or even `Field`?